sdk/sqlmanagement/Microsoft.Azure.Management.Sql/tests/ElasticPoolCrudScenarioTests.cs

```diff
@@ -250,6 +291,49 @@ public async Task TestCancelUpdateElasticPoolOperation()
             Assert.Equal(4, epa.Where(a => a.Operation == "UPDATE").Count());
         }
 
+        private void TestUpdateHyperscaleElasticPool<TUpdateModel>(
+           SqlManagementClient sqlClient,
+           ResourceGroup resourceGroup,
+           Server server,
+           Func<TUpdateModel> createModelFunc,
+           Func<string, string, string, TUpdateModel, ElasticPool> updateFunc)
+        {
+            // Create a Hyperscale elasticPool 
+            // 
+            string epName = SqlManagementTestUtilities.GenerateName();
+            var epInput = new ElasticPool()
+            {
+                Location = server.Location,
+                Sku = new Microsoft.Azure.Management.Sql.Models.Sku("HS_Gen5_4")
+            };
+            var returnedEp = sqlClient.ElasticPools.CreateOrUpdate(resourceGroup.Name, server.Name, epName, epInput);
+            SqlManagementTestUtilities.ValidateElasticPool(epInput, returnedEp, epName);
+            var epa = sqlClient.ElasticPoolActivities.ListByElasticPool(resourceGroup.Name, server.Name, epName);
+            Assert.NotNull(epa);
+            Assert.Equal(1, epa.Count());
+            Assert.Equal(1, epa.Where(a => a.Operation == "CREATE").Count());
+
+            // Verify pool has default HighAvailabilityReplicaCount
+            Assert.Equal(1, returnedEp.HighAvailabilityReplicaCount);
+
+            // Update HighAvailabilityReplicaCount
+            // 
+            dynamic epInput2 = createModelFunc();
+            epInput2.HighAvailabilityReplicaCount = 2;
+
+            returnedEp = updateFunc(resourceGroup.Name, server.Name, epName, epInput2);
+            SqlManagementTestUtilities.ValidateElasticPool(epInput2, returnedEp, epName);
+            epa = sqlClient.ElasticPoolActivities.ListByElasticPool(resourceGroup.Name, server.Name, epName);
+            Assert.NotNull(epa);
+            Assert.Equal(2, epa.Count());
+            Assert.Equal(1, epa.Where(a => a.Operation == "CREATE").Count());
+            Assert.Equal(1, epa.Where(a => a.Operation == "UPDATE").Count());
+
+            // Verify pool has updated HighAvailabilityReplicaCount
+            // 
+            Assert.Equal(2, returnedEp.HighAvailabilityReplicaCount);
```

Comment:
I understand that this test covers:
1. Create EP with default replica count and assert
2. Update EP replica count and assert

I think need to add following tests:
3. Create EP with specified replica count and assert
4. Update EP without changing replica count and assert replica count not changed
5. Create pooled VLDB without specifying replica count (because not supported to set on db level when pooled) replica count should match pool's replica count
6. Move pooled VLDB out of pool without specifying replica count (because not supported on db level when pooled) should keep the replica count of source pool.
7. Get EP should return correct replica count (specifically test GET, rather than checking the returned object from create/update)
8. Get pooled VLDB should return correct replica count matching pool's specifically test GET, rather than checking the returned object from create/update)

Reply:
I added the additional testing scenarios, thanks for the recommendations!